src/services/completions.ts

```diff
@@ -3663,71 +3664,131 @@ function getCompletionData(
+                        exportKind: ExportKind.Named,
+                        moduleFileName: fileName,
+                        isFromPackageJson: false,
+                        moduleSymbol,
+                        symbol: firstAccessibleSymbol,
+                        targetFlags: skipAlias(firstAccessibleSymbol, typeChecker).flags,
+                    }],
+                    position,
+                    isValidTypeOnlyAliasUseSite(location),
+                ) || {};
+
+                if (moduleSpecifier) {
+                    const origin: SymbolOriginInfoResolvedExport = {
+                        kind: getNullableSymbolOriginInfoKind(SymbolOriginInfoKind.SymbolMemberExport),
+                        moduleSymbol,
+                        isDefaultExport: false,
+                        symbolName: firstAccessibleSymbol.name,
+                        exportName: firstAccessibleSymbol.name,
+                        fileName,
+                        moduleSpecifier,
+                    };
+                    symbolToOriginInfoMap[index] = origin;
                 }
-                addSymbolOriginInfo(symbol);
-                addSymbolSortInfo(symbol);
-                symbols.push(symbol);
             }
         }
-        else {
+
+        function addLiteralSymbol() {
             addSymbolOriginInfo(symbol);
             addSymbolSortInfo(symbol);
             symbols.push(symbol);
         }
 
+        /** Combines a property access expression (from the completion location to the left-most name in the computed property expression)
+         * and the computed property expression itself to return an expression to access the computed property from the completion location.
+         * Assumes that the left-most symbol is accessible so there should always be a way to access the computed property symbolically. */
+        function createComputedPropertyAccess(nameSymbol: Symbol, leftMostNameSymbol: Symbol, computedPropertyNameExpression: Expression) {
+            let node: Node | undefined;
+            if (!isTransientSymbol(nameSymbol)) {
+                node = typeChecker.symbolToEntityName(nameSymbol, /*meaning*/ undefined!, contextToken, NodeBuilderFlags.UseAliasDefinedOutsideCurrentScope);
+            }
+            else {
+                // Object literals assigned as const
+                const leftMostNodeAccessExpression = typeChecker.symbolToNode(leftMostNameSymbol, /*meaning*/ undefined!, contextToken, NodeBuilderFlags.UseAliasDefinedOutsideCurrentScope);
+                type OnlyPropertyAccess = Identifier | (PropertyAccessExpression & { expression: OnlyPropertyAccess; });
+                node = createPropertyAccess(computedPropertyNameExpression as OnlyPropertyAccess);
+                function createPropertyAccess(n: OnlyPropertyAccess): Expression {
+                    if (isIdentifier(n)) {
+                        return leftMostNodeAccessExpression! as Expression; //TODO ! and cast

```

Comment:
Not sure what to do here. The factory wants `Expression` but the base case here is coming from `symbolToNode` which is going to be a `Node`

Reply:
You could `Debug.assertNode` just to make sure.